lambda_functions/va_profile/va_profile_opt_in_out_lambda.py

```diff
@@ -124,23 +155,26 @@ def va_profile_opt_in_out_lambda_handler(event: dict, context) -> dict:
 
         try:
             params = (                             # Stored function parameters:
-                record["VaProfileId"],             #     _va_profile_id
-                record["CommunicationItemId"],     #     _communication_item_id
-                record["CommunicationChannelId"],  #     _communication_channel_name
+                record["vaProfileId"],             #     _va_profile_id
+                record["communicationItemId"],     #     _communication_item_id
+                record["communicationChannelId"],  #     _communication_channel_name
                 record["allowed"],                 #     _allowed
                 record["sourceDate"],              #     _source_datetime
             )
 
             if db_connection is None or db_connection.status != 0:
                 # Attempt to (re-)establish a database connection
-                db_connection = make_connection()
+                db_connection = make_connection(worker_id)
 
             if db_connection is None:
                 raise RuntimeError("No database connection.")
 
             # Execute the stored function.
             with db_connection.cursor() as c:
-                put_record["status"] = "COMPLETED_SUCCESS" if c.execute(OPT_IN_OUT_QUERY, params) else "COMPLETED_NOOP"
+                # https://www.psycopg.org/docs/cursor.html#cursor.execute
+                c.execute(OPT_IN_OUT_QUERY, params)
+                put_record["status"] = "COMPLETED_SUCCESS" if c.fetchone()[0] else "COMPLETED_NOOP"
```

Comment:
@k-macmillan This code seems to be doing what I intended.  I ran some queries using Python manually on the ci_test_ci container:

```
>>> import psycopg2
>>> connection = psycopg2.connect("postgresql://postgres:LocalPassword@db:5432/test_notification_api_master")
>>> with connection.cursor() as c:
...     c.execute("select * from va_profile_local_cache;")
...     one = c.fetchone()
...     print(type(one))
...     print(one)
... 
<class 'tuple'>
(177, 1, 1, False, 1, datetime.datetime(2022, 4, 7, 19, 37, 59, 320000))
```

This is the query the lambda executes to call the stored function:

```
>>> with connection.cursor() as c:
...     #c.execute("select * from va_profile_local_cache;")
...     c.execute("SELECT va_profile_opt_in_out(2, 2, 2, True, '2022-03-07T19:37:59.320Z');")
...     one = c.fetchone()
...     print(type(one))
...     print(one)
... 
<class 'tuple'>
(True,)
```

This is the output I expect.  As `fetchone` returns a tuple, I cannot reference by column name.

If you want to reproduce this test, start and manually network containers as described in tests/README.md.

Reply:
With `rowcount`:

```
>>> with connection.cursor() as c:
...     c.execute("select * from va_profile_local_cache;")
...     one = c.fetchone()
...     print(type(one))
...     print(one)
... 
<class 'tuple'>
(177, 1, 1, False, 1, datetime.datetime(2022, 4, 7, 19, 37, 59, 320000))
>>> c.rowcount
1
>>> with connection.cursor() as c:
...     c.execute("select va_profile_opt_in_out(1, 1, 1, True, '2022-01-07T19:37:59.320Z');")
...     one = c.fetchone()
...     print(type(one))
...     print(one)
...     print(c.rowcount)
... 
<class 'tuple'>
(False,)
1
```

The stored function does not update anything because the date is older than the existing date, but rowcount=1 because the call to a stored function is a select statement.  This is the number of rows returned rather than the number of rows affected.